### PR TITLE
Library for manipulating grib1 and grib2 files

### PR DESCRIPTION
**Pull request by robertdfrench**
Needed for building grib support into CDO and some other climate tools.

### CODE REVIEW

File: var/spack/repos/builtin/packages/grib_api/package.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+class GribApi(Package):
+    """The ECMWF GRIB API is an application program interface accessible from C,
+       FORTRAN and Python programs developed for encoding and decoding WMO FM-92
+       GRIB edition 1 and edition 2 messages."""
+
+    homepage = "https://software.ecmwf.int/wiki/display/GRIB/Home"
+    url      = "https://software.ecmwf.int/wiki/download/attachments/3473437/grib_api-1.17.0-Source.tar.gz?api=v2"
+
+    version('1.17.0', 'bca7114d2c3100501a08190a146818d2')
+    depends_on('netcdf~mpi')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/grib_api/package.py]: Does this mean `grib_api` cannot be built with `netcdf+mpi`?

**Comment by adamjstewart** [comment]
You should probably change the package name from `grib_api` to `grib-api` as that is what most Spack packages use. We should really fix `spack create` to do this by default...

**Comment by adamjstewart** [comment]
Also, can you fix the flake8 warning? Looks like the only problem is that line 30 is one character too long.

File: var/spack/repos/builtin/packages/grib_api/package.py (hunk 1 above)
Comment (robertdfrench) [inline var/spack/repos/builtin/packages/grib_api/package.py]: Ah, no, it means I did not want it to build with netcdf+mpi and I was not careful to consider the needs of others. I will revise this to use an `mpi` variant which is off by default.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/grib_api/package.py]: If you don't need the mpi variant for any purpose other than telling netcdf to use the mpi variant, you should leave it out. Just say that `grib_api` depends on `netcdf`, and if you want to build netcdf a particular way, you can either specify that in your `packages.yaml` file or by hand with:
```console
$ spack install grib_api ^netcdf~mpi
```

**Comment by adamjstewart** [comment]
My comment got hidden by the previous push, so copying it here for easier viewing:


If you don't need the mpi variant for any purpose other than telling netcdf to use the mpi variant, you should leave it out. Just say that `grib-api` depends on `netcdf`, and if you want to build netcdf a particular way, you can either specify that in your `packages.yaml` file or by hand with:
```
$ spack install grib-api ^netcdf~mpi
```

**Comment by robertdfrench** [comment]
that's brilliant. Yes, that way variants don't become contagious. Thank you!

**Comment by robertdfrench** [comment]
@adamjstewart Does that latest commit LGTY?

**Comment by adamjstewart** [comment]
Yep, LGTM, thanks!

**Pull request merged**